sympy/matrices/matrices.py

```diff
@@ -1549,11 +1549,12 @@ def jordan_cell_power(jc, n):
             l = jc[0, 0]
             if (num < 0) == True and l == 0:
                 raise ValueError("Matrix det == 0; not invertible")
-            if (num < 2) == True:
-                b = self.jordan_cells()[1]
-                for i in range(len(b)):
-                    if (b[i]).rows > 1 and b[i][0, 0] == 0:
-                        raise ValueError("Cannot be evaluated")
+            if not isinstance(num, (int, Integer, Expr)):
+                if not num - int(num) == 0:
```

Comment:
`num.is_integer` will work only if `num` has been sympified. I think that should be done in `__pow__` *before* calling `_matrix_pow_by_jordan_blocks`. Then `num` is not an `int` or a `float`. A `num` of class `Expr` could probably be allowed, if `num.is_integer` is true.